addons/website_sale/controllers/main.py

```diff
@@ -1081,19 +1081,25 @@ def shop_address_submit(
         else:
             callback = callback or '/shop/checkout'
 
+        is_anonymous_cart = order_sudo._is_anonymous_cart()
         partner_sudo, feedback_dict = self._create_or_update_address(
             partner_sudo,
             address_type=address_type,
             use_delivery_as_billing=use_delivery_as_billing,
             callback=callback,
             order_sudo=order_sudo,
+            # Skip required fields check for customers' main address when order only contains
+            # service type products
+            skip_address_required_fields=(
+                order_sudo.only_services
+                and (partner_sudo == order_sudo.partner_id or is_anonymous_cart)
+            ),
```

Comment:
Doesn't work well because the `/shop/confirm_order` route will still redirect you to the address page if something is missing :eyes:.
So you can save, but you land on the address page again because something is missing, but without any feedback.